plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/save-hub.tsx

```diff
@@ -179,8 +183,27 @@ export const SaveHub = () => {
 			return (
 				<Button
 					variant="primary"
+					isBusy={ isResolving }
 					onClick={ () => {
-						sendEvent( 'FINISH_CUSTOMIZATION' );
+						const homeUrl: string = getSetting( 'homeUrl', '' );
+						// Wait for 5 seconds before redirecting to the transitional page. This is to ensure that the site preview image is refreshed.
+						const WAIT_BEFORE_REDIRECT = 5000;
+
+						setIsResolving( true );
+						// Pre-fetch the site preview image for the site for transitional page.
+						window
```

Comment:
This logic should be in the xstate machine definition, you reckon?

The pattern I used in core-profiler was just to toggle it to show the spinner in the button then add a state after that does the calls etc and then only update the meta component after those states have finished.

In the case of this we can do a state that does the fetch, and then `after: 5000` it goes to the transitional screen state

Reply:
Thanks for the suggestion, it sounds good! I'll move this into XState.

Reply:
Updated in 58a5546fdee359b600b4da40c2b8c081342634b1